big_tests/tests/sm_SUITE.erl

```diff
@@ -591,6 +610,36 @@ resend_unacked_after_resume_timeout(Config) ->
     escalus_connection:stop(Bob),
     escalus_connection:stop(NewAlice).
 
+ping_timeout(Config) ->
+    logger_ct_backend:start(),
+    logger_ct_backend:capture(error),
+
+    %% connect Alice and simulate ping timeout by dropping connection
+    Alice = connect_fresh(Config, alice, sr_presence),
+
+    escalus_client:wait_for_stanza(Alice),
+    ct:sleep(?PING_REQUEST_TIMEOUT + ?PING_INTERVAL + timer:seconds(1)),
+
+    %% attempt to resume the session after the connection drop
+    NewAlice = sm_helper:kill_and_connect_with_resume_session_without_waiting_for_result(Alice),
+
+    %% after resume_timeout, we expect the session to be closed
+    escalus_connection:get_stanza(NewAlice, failed_resumption),
+
+    %% bind a new session and expect unacknowledged messages to be resent
+    escalus_session:session(escalus_session:bind(NewAlice)),
+    send_initial_presence(NewAlice),
+
+    % check if the error occurred
+    FilterFun = fun(_, Msg) ->
+                    re:run(Msg, "parse_iq_id_failed", [global]) /= nomatch
+                end,
+    ?assertEqual([], logger_ct_backend:recv(FilterFun)),
```

Comment:
checking that some error hasn't occured is a weak approach, especiallly using regular expressions, parse_iq_id_failed identifier may change in the future and this testcase will become false-positive.

I think mocking on the MIM node is a better to make a validation.